### DISCUSSION TITLE
Cannot match primary target no matter what

### DISCUSSION THREAD

**Discussion by ClayCore** [unpoly/unpoly#671] [Q&A] [answered]
I've got the following html, rendered with eta on a deno server.
```html
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <title>Redirect test</title>

    <link rel="stylesheet" href="/css/main.css" />

    <script src="/js/main.js" type="module"></script>

    <script src="https://cdn.jsdelivr.net/npm/unpoly@3.9.1/unpoly.min.js"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/unpoly@3.9.1/unpoly.min.css" />
  </head>
  <body>
    <div id="root">
      <h1>Hello world!</h1>

      <main>
        <form action="/redirect" method="post" up-target=".content">
          <label for="target">Select destination:</label>
          <select name="target">
            <option value="/1">First</option>
            <option value="/2">Second</option>
          </select>
          <button type="submit">Send</button>
        </form>
        <div class="content"></div>
      </main>
    </div>
  </body>
</html>
```

The route works and renders either `/1.eta` or `/2.eta` into small fragments with just a test heading that I wish to insert into `.content`. All responses are 200 OKs and I can see in the devTools that the response is the correct html.

However, unpoly refuses to acknowledge that `.content` exists in the original HTML, and not just that, because I tried `main` and other selectors as well. It **always** fails and fallbacks into injecting the fragment into body. 

Here's a screenshot of the devTools with logging enabled. I logged `document.querySelector('.content')` just to make sure i'm not going crazy, and no surprises, it exists.
![image](https://github.com/user-attachments/assets/4a9227d6-9c24-46f0-8ad5-ce67deae4165)

I tried various other attributes like `up-layer` to no avail. No idea what I'm doing wrong to be perfectly honest.

I can provide all the backend code if necessary

**Comment by triskweline** [comment, marked as answer]
> However, unpoly refuses to acknowledge that .content exists in the original HTML

`.content` must exist in both the current page and the response.

> I can provide all the backend code if necessary

To help us help you, please provide a Codepen or Glitch project that reproduces your problem. Here is a Glitch template that loads Unpoly for you: https://glitch.com/edit/#!/empty-unpoly-project

**Reply by ClayCore** [reply]
> `.content` must exist in both the current page and the response.

This actually fixed it. I didn't notice it in docs but it does appear to work like blocks in templating engines. It's fixed and works perfectly fine now. Thank you!
